pagefaultgames/pokerogue
# [Enhancement] [UI/UX] Add ability and passive tooltips to starter select screen

## What are the changes the user will see?
On the starter select screen, hovering the cursor over a Pokémon's Ability or Passive will show a tooltip with that Ability's description. Pressing the Toggle Ability key while the tooltip is visible will update its text.

Selecting a different Pokémon or deselecting one altogether will hide any visible tooltip.

## Why am I making these changes?
There is no way to view the details of the selected Ability and Passive on the starter select screen. Move details can be viewed in the "Manage Moves" menu, and Natures are simple enough that even without the menu, their impact is clear. However, Abilities and Passives are opaque and require a visit to an external website if a player is not already familiar with their effect.

Since Abilities are a short list of three and Passives are single values, creating a menu to view these details seemed like overkill. A tooltip is a simple way to get the information on the screen

## What are the changes from a developer perspective?
The change mostly just leverages the existing tooltip system, calling `setInteractive` on `this.pokemonAbilityText` and `this.pokemonPassiveText`. A mock for `setInteractive` was also added to mockText, to keep the tests passing.

New tooltip management functions `getTooltip` and `editTooltip` have been added, to support the case of the Toggle Ability key. As the tooltip is shown in an event handler, having the ability to edit it in place when it is currently visible is essential for this task. Knowing its current visibility and text state is also useful, as without this information, the egg candy count tooltip would also updated to show the Ability text when toggling through the Ability list.

Tooltips will also now reverse on the y-axis if they would fall off the bottom of the screen, and their maximum width has been increased from 696 to 850.

The egg candy count tooltip previously remained on the screen when changing the selected Pokémon, but it did not update its value. Now, it will disappear, and as before, the user will have to re-hover the sprite to see the new value.

Mobile tooltips are as functional as they ever are.

### Screenshots/Videos
#### Desktop interaction:
![ndiRXr3DVZ](https://github.com/user-attachments/assets/7b87e59d-209f-483c-87f3-ccba3a5ec939)

#### Mobile still:
![Screenshot_20240904_122334_Chrome](https://github.com/user-attachments/assets/39fdc7bc-3ab6-4aaf-9eee-094387702525)


## How to test the changes?
The tests for the starter select screen continue to pass with the updated mock. No new tests were added to ensure the tooltips' presence.

To manually test, simply visit the starter select screen and hover your mouse over a Pokémon's Ability or Passive.

## Checklist
